namespaces/parse-result.md

```diff
@@ -0,0 +1,138 @@
+# Parse Result Namespace
+
+This document defines elements needed for holding the result of parsing
+(or other processing) API description documents.
+
+This document uses [API Namespace][].
+
+## Content
+
+<!-- TOC depth:3 withLinks:1 updateOnSave:0 -->
+- [Parse Result Namespace](#parse-result-namespace)
+	- [Content](#content)
+	- [About this Document](#about-this-document)
+	- [Parse Result (Element)](#parse-result-element)
+		- [Properties](#properties)
+		- [Example](#example)
+	- [Annotation (Element)](#annotation-element)
+		- [Properties](#properties)
+		- [Example](#example)
+	- [Source Map (Element)](#source-map-element)
+		- [Properties](#properties)
+		- [Example](#example)
+<!-- /TOC -->
+
+## About this Document
+
+This document conforms to [RFC 2119][], which says:
+
+> The key words “MUST”, “MUST NOT”, “REQUIRED”, “SHALL”, “SHALL NOT”, “SHOULD”, “SHOULD NOT”, “RECOMMENDED”, “MAY”, and “OPTIONAL” in this document are to be interpreted as described in RFC 2119.
+
+[MSON][] is used throughout this document.
+
+## Parse Result (Element)
+
+A result of parsing of an API description document.
+
+### Properties
+
+- `element`: parseResult (string, fixed)
+- `content` (array)
+    - (Category)
+    - (Annotation)
+
+### Example
+
+Given following API Blueprint document:
+
+```apib
+# GET /1
+```
+
+The parse result is (using null in `category` content for simplicity):
+
+```json
+["parseResult", {}, {}, [
+    ["category", {"class": "api"}, {"sourceMap": [[0,9]]}, null],
+    ["annotation", {"class": "warning"}, { "code": 6, "sourceMap": [[0,9]] }, "action is missing a response"]
+  ]
+]
+```
+
+## Annotation (Element)
+
+Annotation for a source file. Usually generated by a parser or adapter.
+
+### Properties
+
+- `element`: annotation (string, fixed)
+- `meta`
+  - `class` (array, fixed)
+      - (enum[string])
+          - error - Annotation represents an error
+          - warning - Annotation represents a warning
+
+- `attributes`
+    - `code` (number) - Parser-specific code of the annotation.
+    Refer to parser documentation for explanation of the codes.
+
+    - `sourceMap` (Source Map) - Location of the annotation in the source file.
+
+- `content` (string) - Textual annotation.
+
+    This is – in most cases – a human-readable message to be displayed to user.
+
+### Example
+
+```json
+{
+  "element": "annotation",
+  "meta": {
+    "class": "warning"
```

Comment:
`meta.class` is an array in refract? Otherwise I think we should have it as string.